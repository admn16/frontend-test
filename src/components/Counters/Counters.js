import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import Card from 'components/Card/Card';
import Counter from 'components/Counter/Counter';

const StyledCounters = styled(Card)`
  margin: 10px 0 !important;
`;

const Counters = ({
  counters,
  onDelete,
  onDecrement,
  onIncrement,
}) => (
  <StyledCounters>
    <article>
      { counters.map(({ id, title, count }) => (
        <Counter
          key={id}
          title={title}
          count={count}
          onIncrement={onIncrement(id)}
          onDelete={onDelete(id)}
          onDecrement={onDecrement(id)}
        />
      )) }
    </article>
  </StyledCounters>
);

Counters.propTypes = {
  counters: PropTypes.arrayOf(PropTypes.object).isRequired,
  onDelete: PropTypes.func.isRequired,
  onDecrement: PropTypes.func.isRequired,
  onIncrement: PropTypes.func.isRequired,
};

export default Counters;
