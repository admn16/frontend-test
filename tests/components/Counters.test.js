import React from 'react';
import renderer from 'react-test-renderer';
import Counters from 'components/Counters/Counters';

describe('<Counters/>', () => {
  const mockFn = () => jest.fn();

  const props = {
    counters: [{ count: 1, title: 'test123', id: '123asd' }],
    onDelete: mockFn,
    onDecrement: mockFn,
    onIncrement: mockFn,
  };

  test('renders correctly', () => {
    const tree = renderer
      .create(<Counters {...props} />)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
