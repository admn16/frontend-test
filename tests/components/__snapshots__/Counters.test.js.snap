// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Counters/> renders correctly 1`] = `
<div
  className="Counters__StyledCounters-vlq2we-0 iBWZCO Card__StyledCard-hu4gdg-0 hNyZiT"
>
  <article>
    <div
      className="Counter__StyledCounter-sc-1q55mxl-0 eWqOki"
    >
      <button
        className="Counter__Delete-sc-1q55mxl-3 foqxoZ"
        onClick={[MockFunction]}
        type="button"
      >
        <i
          className="fas fa-trash-alt"
        />
      </button>
      <span
        className="Counter__Title-sc-1q55mxl-1 gzIbEF"
      >
        test123
      </span>
      <button
        className="Counter__Increment-sc-1q55mxl-4 gSvDCI Counter__Delete-sc-1q55mxl-3 foqxoZ"
        onClick={[MockFunction]}
        type="button"
      >
        <i
          className="fas fa-plus"
        />
      </button>
      <span
        className="Counter__Count-sc-1q55mxl-2 egAVVf"
      >
        1
      </span>
      <button
        className="Counter__Decrement-sc-1q55mxl-5 cJKjoF Counter__Increment-sc-1q55mxl-4 gSvDCI Counter__Delete-sc-1q55mxl-3 foqxoZ"
        onClick={[MockFunction]}
        type="button"
      >
        <i
          className="fas fa-minus"
        />
      </button>
    </div>
  </article>
</div>
`;
